django/http/request.py

```diff
@@ -628,6 +633,20 @@ def bytes_to_text(s, encoding):
         return s
 
 
+def _parse_host_header(host_header):
+    """
+    Returns a (domain, port) tuple for a given host.
+
+    Neither domain name nor port are validated.
+    """
+
+    if host_header[-1] == ']':
+        # It's an IPv6 address without a port.
+        return host_header, ''
+    bits = host_header.rsplit(':', 1)
+    return tuple(bits) if len(bits) == 2 else (bits[0], '')
```

Comment:
TODO: We could raise SupiciousOperation here if the port is empty (ie the hostname ended with `:`)

Reply:
since there's no validation here and `get_port()` is relying on this code I would think you'd need to consider other malformed cases, such as `example.com:abc` or `1.1.1.1:443]`

Reply:
Yeah I am leaning toward calling `int` on the port part as only validation.

Reply:
`isdigit` is better (see https://github.com/django/django/pull/12679/files#diff-0eb6c5000a61126731553169fddb306eR174 ) `int` is slow and also will take `-443` to be valid ;)

Reply:
Oh, I forgot about that one, let me push a fix for this.

Reply:
Mhm, this is tricky. I am not exactly sure on which level to insert `isdigit` :( I am leaning towards doing it in `get_host` & `get_port` since `get_raw_uri` should not validate it. I also just realized that the `host` returned from `_parsed_host_header` is never used, only the `port` & `reconstructed` one -- so I am wondering if I can get away by just returning `reconstructed` *scratches head*.